M365/src/DeleteContactsFromFolder/DeleteContactsFromFolder.ps1

```diff
@@ -0,0 +1,188 @@
+                            Method             = "DELETE"
+                            ExpectedStatusCode = 204
+                            GraphApiUrl        = $graphApiUrl
+                        }
+                        $deleteContactResponse = Invoke-GraphApiRequest @deleteContactParams
+                        if ($deleteContactResponse.Successful -eq $false) {
+                            Write-Host "Unable to delete the contact" -ForegroundColor Red
+                            $unableToDeleteAllContacts = $true
+                        } else {
+                            Write-Host "Deleted contact successfully" -ForegroundColor Green
+                        }
+                    }
+
+                    if ($unableToDeleteAllContacts -eq $false) {
+                        Write-Host "Deleted folder content, now deleting the folder itself..." -ForegroundColor Cyan
+                        # Remove the folder itself
+                        $deleteContactFolderParams = @{
+                            Query              = "me/contactFolders/$($folderObj.id)"
+                            AccessToken        = $token.AccessToken
+                            Method             = "DELETE"
+                            ExpectedStatusCode = 204
+                            GraphApiUrl        = $graphApiUrl
+                        }
+                        $deleteContactFolderResponse = Invoke-GraphApiRequest @deleteContactFolderParams
+                        if ($deleteContactFolderResponse.Successful -eq $false) {
+                            Write-Host "Unable to delete the folder: '$($folderObj.displayName)'" -ForegroundColor Red
+                            Write-Host "Id: Id: $($folderObj.id)" -ForegroundColor Red
+                            exit
+                        }
+                        Write-Host "Folder: '$($folderObj.displayName)' has been deleted" -ForegroundColor Green
+                    } else {
+                        Write-Host "Unable to delete all contacts in the folder. Please run the script again" -ForegroundColor Yellow
+                    }
+                }
+            } else {
+                Write-Host "A folder with the name '$folderToBeDeleted' wasn't found" -ForegroundColor Red
+            }
+        } else {
+            Write-Host "No Folders exists in this mailbox - nothing to do" -ForegroundColor Yellow
+        }
+    } catch {
+        Write-Host "An error occurred. Please contact support"
+    }
+}
+
+try {
+    $loggerParams = @{
+        LogName        = "DeleteContactsFromFolder"
+        LogDirectory   = (Get-Location).Path
+        AppendDateTime = $true
+        ErrorAction    = "SilentlyContinue"
+    }
+
+    $Script:Logger = Get-NewLoggerInstance @loggerParams
+    SetProperForegroundColor
+    SetWriteHostAction ${Function:Write-DebugLog}
+    SetWriteVerboseAction ${Function:Write-DebugLog}
+
+    Main
+} finally {
```

Comment:
Should include a catch to try to pull out error information to assist with debugging issues. Include `Write-HostErrorInformation` or `Write-VerboseErrorInformation` from `Shared\Write-ErrorInformation.ps1`

Reply:
Done.